llama-index-integrations/retrievers/llama-index-retrievers-you/llama_index/retrievers/you/base.py

```diff
@@ -34,3 +34,28 @@ def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
 
         search_hits = ["\n".join(hit["snippets"]) for hit in results["hits"]]
         return [NodeWithScore(node=TextNode(text=s), score=1.0) for s in search_hits]
+
+
+class YouNewsRetriever(BaseRetriever):
+    """You retriever."""
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> None:
+        """Init params."""
+        self._api_key = api_key or os.environ["YOU_API_KEY"]
+        super().__init__(callback_manager)
+
+    def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
+        """Retrieve."""
+        headers = {"X-API-Key": self._api_key}
+        results = requests.get(
+            f"https://api.ydc-index.io/news?q={query_bundle.query_str}",
+            headers=headers,
+        ).json()
+
+        news_hits = [f'{news["description"]}\nURL: {news["url"]}' for news in results["news"]["results"]]
+        print(news_hits)
```

Comment:
Adding logging is advisable if you intend to print it; otherwise, it's recommended to remove it.